### ISSUE TITLE
ERC20 transfer violations due to suspected rounding errors

### ISSUE THREAD

**Issue by tomlinton** [OriginProtocol/origin-dollar#415]
From branch `tomlinton/echidna-action`:

```
crytic_self_transferFrom_to_other_ERC20PropertiesTransferable: failed!💥
  Call sequence:
    initialize("\145s\".\245\154\US\vn\SUB\216\239\DC1\133[`\DC1\223","\SO\&HEO\157",0xf17f52151ebef6c7334fad080c5704d77216b732) from: 0x627306090abab3a6e1400e9345bc60c78a8bef57 Time delay: 113292 seconds Block delay: 22491
    mint(0xf17f52151ebef6c7334fad080c5704d77216b732,112) from: 0xf17f52151ebef6c7334fad080c5704d77216b732 Time delay: 54309 seconds Block delay: 6289
    changeSupply(157198258) from: 0xf17f52151ebef6c7334fad080c5704d77216b732 Time delay: 257855 seconds Block delay: 3948
```

```
  Call sequence:
    initialize("\145s\".\245\154\US\vn\SUB\216\239\DC1\133[`\DC1\223","\SO\&HEO\157",0xf17f52151ebef6c7334fad080c5704d77216b732) from: 0x627306090abab3a6e1400e9345bc60c78a8bef57 Time delay: 113292 seconds Block delay: 22491
    mint(0xf17f52151ebef6c7334fad080c5704d77216b732,112) from: 0xf17f52151ebef6c7334fad080c5704d77216b732 Time delay: 54309 seconds Block delay: 6289
    changeSupply(49) from: 0xf17f52151ebef6c7334fad080c5704d77216b732
```

Still investigating the root cause but it is likely due to rounding errors. Note that not every echidna run will generate these. Sometimes I get a clean run.

**Comment by tomlinton** [comment]
One potential way to fix this is to recalculate _totalSupply after the rebasingCreditsPerToken has been calculated

I have to think through the impact of that, one obvious consequence is that _totalSupply will no longer be set to exactly the _newTotalSupply argument to changeSupply.

**Comment by tomlinton** [comment]
Closed by #418
